src/DIRAC/WorkloadManagementSystem/Utilities/RemoteRunner.py

```diff
@@ -0,0 +1,130 @@
+""" RemoteRunner
+
+Mostly called by workflow modules, RemoteRunner is generally the last component to get through before
+the script/application execution on a remote machine.
+Depending on an environment variable WORKLOADEXECLOCATION, it decides whether it should take care of the execution.
+RemoteRunner has been designed to send script/application on remote worker nodes having no outbound connectivity
+(e.g. supercomputers)
+"""
+import os
+import shlex
+import time
+
+from DIRAC import gLogger, gConfig, S_OK, S_ERROR
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+from DIRAC.Resources.Computing.ComputingElementFactory import ComputingElementFactory
+from DIRAC.ConfigurationSystem.Client.Helpers.Resources import getQueue
+from DIRAC.WorkloadManagementSystem.Client import PilotStatus
+
+
+class RemoteRunner(object):
+    def __init__(self):
+        self.log = gLogger.getSubLogger("RemoteRunner")
+        self.workloadExecutionLocation = gConfig.getValue("/LocalSite/WorkloadExecution")
+
+    def is_remote_execution(self):
+        """Main method: decides whether the execution will be done locally or remotely via a CE.
+
+        :return: bool
+        """
+
+        # if workloadExecutionLocation is not empty, this means the workload should be executed
+        # in a different remote location. This mainly happens when the remote Site has no
+        # external connectivity and can only execute the workload itself.
+        return bool(self.workloadExecutionLocation)
+
+    def execute(self, command):
+        """Execute the command remotely via a CE
+
+        :param str command: command to execute remotely
+        """
+        # Set up Application Queue
+        self.log.verbose("Remote application execution on:", self.workloadExecutionLocation)
+        result = self._setUpworkloadCE(self.workloadExecutionLocation)
+        if not result["OK"]:
+            return result
+        workloadCE = result["Value"]
+
+        # Add the command in an executable file
+        executable = self._wrapCommand(command)
+        # get inputs file from the current working directory
+        inputs = os.listdir(".")
+        inputs.remove(os.path.basename(executable))
+        self.log.verbose("The executable will be sent along with the following inputs:", ",".join(inputs))
+        # request the whole directory as output
+        outputs = ["/"]
+
+        # Submit the command as a job
+        result = workloadCE.submitJob(executable, workloadCE.proxy, inputs=inputs, outputs=outputs)
+        if not result["OK"]:
+            return S_ERROR("Cannot submit the command")
```

Comment:
Why not `return result` ?

Reply:
Same question for all other cases in this module.

Reply:
You are right, I've just changed it